aws/onboarding/byok8s/python/rubrik_eks_cluster_private.py

```diff
@@ -0,0 +1,1203 @@
+        )
+
+        # 3. Setup launch template
+        launch_template_id = setup_launch_template(
+            aws_manager=aws_manager,
+            cluster_name=cluster_name,
+            certificate_authority=cluster_certificate_authority,
+            cluster_endpoint=cluster_endpoint,
+            region=aws_region,
+            prefix=prefix,
+            instance_profile_arn=worker_node_instance_profile_arn,
+            worker_node_security_group=node_security_group_id,
+            node_type=node_type
+        )
+
+        # 4. Create Auto Scaling group
+        aws_manager.create_autoscaling_group(
+            name=prefix + "-autoscaling-group",
+            launch_template_id=launch_template_id,
+            min_size=1,
+            max_size=64,
+            desired_capacity=1,
+            vpc_zone_identifier=",".join(subnet_ids),
+            cluster_name=cluster_name
+        )
+
+        # 5. Connect Worker nodes to EKS cluster
+        connect_worker_nodes_to_cluster(
+            worker_node_role_arn=worker_node_role_arn,
+            cross_account_role_arn=cross_account_role,
+            cluster_endpoint=cluster_endpoint,
+            cluster_certificate_authority=cluster_certificate_authority,
+            cluster_name=cluster_name,
+            cluster_arn=cluster_arn
+        )
+
+        print("Completed setup for customer exocompute")
+        return str(cluster_arn)
+    except Exception as e:
+        print(f"Error setting up customer exocompute: {str(e)}")
+        raise
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Setup Customer Exocompute')
+    parser.add_argument(
+        '--aws-access-key', required=True, help='AWS Access Key'
+    )
+    parser.add_argument(
+        '--aws-secret-key', required=True, help='AWS Secret Key'
+    )
+    parser.add_argument(
+        '--aws-session-id', required=True, help='AWS Session ID'
+    )
+    parser.add_argument('--aws-region', required=True, help='AWS Region')
+    parser.add_argument('--vpc-id', required=True, help='VPC ID')
+    parser.add_argument(
+        '--jumpbox-security-group-id',
+        required=True,
+        help='Jumpbox Security Group ID'
+    )
+    parser.add_argument(
+        '--subnet-ids', required=True, nargs='+', help='Subnet IDs'
+    )
+    parser.add_argument('--prefix', required=True, help='Prefix')
+    parser.add_argument(
+        '--master-role-arn', required=True, help='Master Role ARN'
+    )
+    parser.add_argument(
+        '--worker-node-role-arn', required=True, help='Worker Node Role ARN'
+    )
+    parser.add_argument(
+        '--worker-node-instance-profile-arn',
+        required=True,
+        help='Worker Node Instance Profile ARN'
+    )
+    parser.add_argument(
+        '--cross-account-role', required=True, help='Cross Account Role ARN'
+    )
+    parser.add_argument('--node-type', default=NODE_TYPE, help='Node Type')
```

Comment:
should we avoid using default here? This is assuming EC2 protection, for S3 we have m5.4xlarge etc...